spec/integration_spec.rb

```diff
@@ -271,4 +271,29 @@ def parse_html(html)
     expect(result).to include("<td>One</td><td>1</td>")
     expect(result).to include("<td>Two</td><td>2</td>")
   end
+
+  it "doesn't inline styles in media queries with features" do
+    document = Roadie::Document.new <<-HTML
+      <html>
+        <head>
+          <link rel="stylesheet" href="/style.css">
+        </head>
+        <body>
+          <div class="colorful"></div>
+        </body>
+      </html>
+    HTML
+
+    document.asset_providers = TestProvider.new(
+      "/style.css" => <<-CSS
+        .colorful { color: green; }
+        @media screen and (max-width 600px) {
+          .colorful { color: red; }
+        }
+      CSS
+    )
+
+    result = parse_html document.transform
+    expect(result).to have_styling('color' => 'green').at_selector('.colorful')
```

Comment:
Please add an expectation that the media query is added to the `<head>`.

You can mark it as `pending` if you cannot make it work and I'll complete that part of the feature.